elmo/elmo-imap4.el

```diff
@@ -465,26 +465,29 @@ COMMAND is a list of command tokens or a single command token."
 (defun elmo-imap4-read-response (session tag)
   "Read parsed response from SESSION.
 TAG is the tag of the command"
-  (with-current-buffer (process-buffer
-                        (elmo-network-session-process-internal session))
-    (while (not (or (string= tag elmo-imap4-reached-tag)
-                    (elmo-imap4-response-bye-p elmo-imap4-current-response)
-                    (when (elmo-imap4-response-garbage-p
-                           elmo-imap4-current-response)
-                      (message "Garbage response: %s"
-                               (elmo-imap4-response-value
-                                elmo-imap4-current-response
-                                'garbage))
-                      t)))
-      (when (memq (process-status
-                   (elmo-network-session-process-internal session))
-                  '(open run))
-        (accept-process-output (elmo-network-session-process-internal session)
-                               1)))
-    (elmo-imap4-debug "[%s] => %s" (format-time-string "%T") (prin1-to-string elmo-imap4-current-response))
-    (setq elmo-imap4-parsing nil)
-    (elmo-imap4-mailbox-size-update-maybe session elmo-imap4-current-response)
-    elmo-imap4-current-response))
+  (let ((process (elmo-network-session-process-internal session)))
+    (with-current-buffer (process-buffer process)
+      (while (not (or (string= tag elmo-imap4-reached-tag)
+                      (elmo-imap4-response-bye-p elmo-imap4-current-response)
+                      (when (not (memq (process-status process) '(open run)))
+                        (elmo-network-close-session session)
+                        (signal 'elmo-imap4-bye-error "Connection is broken, session is closed"))
+                      (when (elmo-imap4-response-garbage-p
+                             elmo-imap4-current-response)
+                        (message "Garbage response: %s"
+                                 (elmo-imap4-response-value
+                                  elmo-imap4-current-response
+                                  'garbage))
+                        t)))
+        (when (memq (process-status
+                     (elmo-network-session-process-internal session))
+                    '(open run))
```

Comment:
We already check process status here, I think you can raise a signal here also.

Reply:
I don't understand what you tried, but I mean https://github.com/ikazuhiro/wanderlust/commit/e889c0c50b1a478026f0788443529282abde88d8 , and It seems work.
Please test.

Reply:
@ikazuhiro ah, I got that you mean. You mean to move raising a error down.

Yes, it also work. I close this PR in favore of your.